OsmAnd/src/net/osmand/plus/plugins/aistracker/AisTrackerLayer.java

```diff
@@ -217,7 +267,17 @@ public void onPrepareBufferImage(Canvas canvas, RotatedTileBox tileBox, DrawSett
 
 		if (tileBox.getZoom() >= START_ZOOM) {
 			AisObject.setOwnPosition(getApplication().getLocationProvider().getLastKnownLocation());
+
 			for (AisObject ais : aisObjectList.values()) {
+				if (view.getMapRenderer() != null) {
+					// Calling updateAisRenderData in onPrepareBufferImage is overhead
+					// but it is needed to update directional line points
+					// also there is no zoom animation for directional line
+					// TODO: SUPPORT THIS IN ENGINE
+					ais.updateAisRenderData(getTileView(), this, bitmapPaint);
+					continue;
+				}
```

Comment:
better split into two cycles for V1 & V2

Reply:
Fixed